Update CVs guidance (#195)
WCRP-CMIP/cmip7-guidance: docs/CMIP7/cv_registration.md

Reviewer: ```suggestion
- Your **institution**, which registers your `institution_id`
- **Model documentation** (EMD) components, which registers your `source_id`
- Any **output grids** if they don't already exist in the CVs and are not covered by your EMD registration 
```
Assuming this flowchart is for modelling centres (as the preceding text suggests) then experiments can be removed. Also a reminder about output grids is probably useful.

Also suggesting not to say "organisation" because the Universe has both "organisation" and "institution". Assuming these have distinct meanings in the Universe, maybe better not to use these words ("organisation" and "institution") interchangeably?

Reviewer 2: This is in CV registration, so I don't think we should remove experiments now -- it might be that at some point we separate the guidance to cover MIPs and modelling groups separately.

I've removed reference to "organisation" for now.

Reviewer: ```suggestion

!!! info "Data production overview:"
    Key steps for data preparation and publication readiness are outlined in the slides from [CMIP 2026 workshop WIP session](https://zenodo.org/records/18934629) (slides 30-36).
    
```
This same box appears in Guidance_for_modellers.md and might be a helpful reminder here too.

Reviewer 2: Done

### 2.1 Institution Registration

Register your institution before registering a model.
This is done in two parts:

1. register as an 'institution member' with an associated ID (i.e. DRS name)
2. register for an institution ID (i.e. DRS name) based on the members that should be associated with the institution ID
    - If your institution ID is associated with a consortium, you will have more than one member and an institution ID that differs from those of your members
    - If your institution ID is associated with a single member, you will have one member and an institution ID that is the same as the the member's ID (i.e. DRS name).

**Repository**: [CMIP7-CVs](https://github.com/WCRP-CMIP/CMIP7-CVs) [Registered Content](https://github.com/WCRP-CMIP/CMIP7-CVs/tree/main/institution)

| Form | Link | Required Fields |
|------|------|-----------------|
| Institution member | [Register Institution member](https://github.com/WCRP-CMIP/CMIP7-CVs/issues/new?template=register-institution-member.yml) | Title, ID (i.e. DRS name), description, [ROR](https://ror.org/) |

> | Institution | [Register Institution](https://github.com/WCRP-CMIP/CMIP7-CVs/issues/new?template=register-institution.yml) | Title, ID (i.e. DRS name), description, members |

Author: Note: the forms aren't up yet, but they will be up soon

Register your institution before registering a model.
This is done in two parts:

> 1. register as an 'institution member' with an associated ID (i.e. DRS name)

Reviewer 3: specify what a DRS name is. This may not be intuitive to all users,

Reviewer: ```suggestion
1. register as an 'institution member' with an associated ID (i.e. [DRS name](Global_Attributes.md#4-data-reference-syntax-drs-elements))
```

Reviewer: Can we clarify here what an "institution member" is, as opposed to just an "institution"?

Reviewer 2: I'm just making an extension to this section explain this ready for comms.

2. register for an institution ID (i.e. DRS name) based on the members that should be associated with the institution ID
    - If your institution ID is associated with a consortium, you will have more than one member and an institution ID that differs from those of your members

>     - If your institution ID is associated with a single member, you will have one member and an institution ID that is the same as the the member's ID (i.e. DRS name).

Reviewer: It's unclear to me here how `institution_id` (the required global attribute that modellers need to ensure is registered) relates to consortia and members. Is it the case that:

- `institution_id` is the DRS name identifying a modelling group that is publishing data (e.g. CCCma)
- the institute identified by `institution_id` might belong to some larger entity, a consortia or other institute (e.g. CCCma is part of ECCC, which has a ROR while CCCma does not, but we use "CCCma" as the DRS name in our published CMIP output)

If so I would have thought that there would be two separate steps: first to register the publishing institute (identified in the DRS by `institution_id`), and a second optional step to register the larger entity (consortium, etc) if it is applicable. And the second step would identify one or more `institution_id`'s as belonging to that larger entity.

Maybe I've got it all wrong, but I think many users would probably ask the same question, so can we clarify in the guidance how this relationship between `institution_id` and the rest of it (consortia) is working?

Reviewer 4: Interesting case! It's actually the other way around from what you're describing.

  The model is:

  - An institution member is a single institution that participates in data production (e.g. CNRM, CERFACS, CCCma). Each gets its own ID (i.e. DRS name).
  - An organisation is the entity that actually publishes data. It can map to a single member or group multiple members together.

  Example — consortium case:

  - CNRM → institution member (DRS name: CNRM)
  - CERFACS → institution member (DRS name: CERFACS)
  - These two publish together, so one organisation with two members:
    - CNRM-CERFACS → organisation (DRS name: CNRM-CERFACS)

  Example — single institution (most common case):

  - CCCma → institution member (DRS name: CCCma)
  - CCCma → organisation with one member (DRS name: CCCma)

  The DRS name is what matters in the publishing workflow — it appears everywhere in filenames, paths, and metadata. The rest of the fields inside the
  registration (full name, description, ROR, etc.) are descriptive and can reflect the broader parent entity. So for CCCma, you'd use ECCC's ROR and full
   institutional name inside the registration, while keeping CCCma as the DRS name. No need to register ECCC separately since it doesn't publish data as
  a whole.

  This single-member case will be the most common one — most centres just register as both a member and an organisation with the same DRS name.

Reviewer 4: Adding these examples directly in the guidance could be profitable ?

Reviewer 2: I've tried to do this, and we can resolve any further questions after I merge

> **Repository**: [CMIP7-CVs](https://github.com/WCRP-CMIP/CMIP7-CVs) [Registered content for institutes](https://github.com/WCRP-CMIP/CMIP7-CVs/tree/main/institution), [Registered content for institute members](https://github.com/WCRP-CMIP/WCRP-universe/tree/esgvoc/institution)

Reviewer 3: What about non cmip7 projects?

Reviewer 4: This is the CMIP7 guidance, so we only link to CMIP7 registration forms here. But under the hood, registering through CMIP7-CVs will create the
  institution/members in the WCRP-universe as well — so they'll be usable by any project (as is already the case).

Reviewer 2: We will have to work this out later -- priority now is to ensure that CMIP7 publication can begin.

| Form | Link | Required Fields |
|------|------|-----------------|

> | Institution member | [Register Institution member](https://github.com/WCRP-CMIP/CMIP7-CVs/issues/new?template=register-institution-member.yml) | Title, ID (i.e. DRS name), description, [ROR](https://ror.org/) |

Reviewer 3: What is an institution vs institution member? Make this clear to the users.

Reviewer 4: See response to JamesAnstey's comment on L36.

Reviewer 2: Have done my best with that section

| Institution | [Register Institution](https://github.com/WCRP-CMIP/CMIP7-CVs/issues/new?template=register-institution.yml) | Title, ID (i.e. DRS name), description, members |
 
**Notes**:
- The **acronym** used for the institution id must be unique and cannot be changed once data is published

> - An **ROR** (Research Organisation Registry) identifier is required for traceability. Find yours at [ror.org](https://ror.org)

Reviewer: Related to my comment above, I'm not sure at which "level" a ROR is needed in the CVs. In my case there is a larger entity (ECCC) that has a ROR, but the DRS name (`institution_id`) is "CCCma" which doesn't have a ROR. From the ROR registration guidelines it seems correct that CCCma, as a small group within a much larger organization, should not have a ROR. 

For traceability I think this should be ok, since CCCma belonging to ECCC means that CCCma has a clear connection to an institute that has a ROR. But I'm not sure how to correctly express this relationship via CVs registration. It might be that the existing CVs entries for CCCma are already ok - and that's great if so - but the same question may come up for other groups.

Reviewer 2: I'll leave this for a future discussion -- @JamesAnstey could you raise this as a separate issue?

Reviewer 4: Same thinking as my comment on L36 — the main goal right now is to enable standard publishing. For CCCma, using ECCC's ROR inside the CCCma
  registration works fine.

  The institution model was discussed a long time ago and this isn't the right time to revisit it. Once the dust settles on the current registration
  process, we could redive into more precise institution modelling if needed.

- If you previously did the institution ID registration process, but your institution is not in the CVs, please [raise an issue in the CMIP7-CVs repository](https://github.com/WCRP-CMIP/CMIP7-CVs/issues/new?template=BLANK_ISSUE)

---

### 2.2 Model Registration (source_id)

**Important**: To register a `source_id`, you **must** complete the EMD (Essential Model Documentation) registration process.

Guidance on constructing a source id is provided in the [Source ID Guidance](Source_ID_guidance.md)

**Prerequisites**:
1. EMD registration completed (see 2.4) - including grids, components, and top-level Model

**Repository**: [CMIP7-CVs](https://github.com/WCRP-CMIP/CMIP7-CVs) [Registered Content](https://github.com/WCRP-CMIP/CMIP7-CVs/tree/main/institution) (content populated via [the EMD](https://github.com/WCRP-CMIP/Essential-Model-Documentation/tree/src-data/model))

> Source IDs and grid labels will be registered automatically within the CVs once EMD has been completed.

Reviewer: ```suggestion
Grid labels and Source IDs will be registered automatically within the CVs once the relevant EMD stages have been completed.
```
(since grid labels don't require the full process to complete)

Reviewer 2: Added

## 3. Registration Workflow

### Overall Registration Order

```mermaid
flowchart LR
    A["1. Institution (CMIP7-CVs)"] --> B["2. EMD Registration (Essential-Model-Documentation)"]

>     B --> C["3. source_id (CMIP7-CVs)"]

Reviewer 3: Shall we add a comment saying the source id name needs to match that as the EMD.

Reviewer: I think we should just remove `source_id` since it's not a separate step users have to do, being covered by EMD.

Reviewer 4: Agreed — users don't have to register source_id in CMIP7-CVs. The EMD top-level model registration propagates it directly (semi-automatically) into
  CMIP7-CVs, so no action needed from the user at this step. The flowchart should reflect that it's an automatic outcome of EMD completion, not a
  separate user action.

>     C --> D["4. Experiments (if needed, CMIP7-CVs)"]

Reviewer: This is done by MIPs, no? I'd assumed this flowchart is for modellers - if so, the experiments box should be deleted.

Reviewer 2: Have tried to clarify this.

Reviewer 4: True that experiments are mostly registered by MIPs, but MIPs are also users who could land on this page. I'd keep it, maybe with a note that this step
   is typically done by MIP leads rather than individual modelling centres.

- **Don't wait**: Start EMD registration early - dependencies mean sequential steps
- **Check dependencies**: A Model Component cannot reference a grid that isn't registered yet, a source cannot refer to components that don't exist. 
- **Be precise**: Acronyms and identifiers cannot be changed after data publication
- **Ask for help**: Consult the documentation first, and if needed use the general forms linked below for questions.

---

## 4. After Registration

> Once your data is registered:

Reviewer: ```suggestion
Once your entries are registered:
```

Reviewer 4: 👍

Reviewer 2: Done

1. The **esgvoc** library will include your entries in the next update

> 2. You can use your registered identifiers in CMOR tables

Reviewer: ```suggestion
2. Your registered identifiers will appear as valid entries in the CMIP7 CMOR tables
```
(to avoid potentially giving users the impression that they should be editing the CMOR tables)

Reviewer 2: Picked up in my next commit

Reviewer 4: +1000 😄 
Definitely don't want users thinking they should edit CMOR tables — if they do, it'll fail at QA/QC anyway since validation runs
  against CV entries, not the tables themselves.

> 3. Your data will pass QA/QC validation for these CV fields

Reviewer: ```suggestion
3. Your model output netCDF files will be able to pass QA/QC validation for these CV fields
```

Reviewer 2: Done

Reviewer 4: ```suggestion
3. Your simulation output netCDF files will be able to pass QA/QC validation for these CV fields
```

Either wording works — I'd slightly prefer "simulation output" over "model output" for precision, but no strong opinion

These repositories maintain the information used by the EMD.
Some of the information is automatically translated out into the CV repositories,
but these EMD repositories are not themselves CV repositories.
**Your data is only registered in the CVs if it appears in the CVs repositories above. Registration in the EMD repositories alone is not registration in the CVs.**

| Repository | Content |
|------------|---------|
| [WCRP-constants](https://github.com/WCRP-CMIP/WCRP-constants/) | Constants required to support the EMD |
| [Essential-Model-Documentation](https://github.com/WCRP-CMIP/Essential-Model-Documentation/) | EMD components |
---

## 6. Getting Help

- **Question about a specific CV value**: Open a [CV value issue](https://github.com/WCRP-CMIP/CMIP7-CVs/issues/new?template=cv-value.md) in the CMIP7 CVS repository
- **CV discussions**: See [Discussions on the CVs repo](https://github.com/WCRP-CMIP/CMIP7-CVs/discussions)
- **General questions**: Open a [blank issue](https://github.com/WCRP-CMIP/CMIP7-CVs/issues/new?template=BLANK_ISSUE) in the CMIP7 CVS repository and ask your question. Please include as much context and other helpful information as possible.

> - **Contact IPO**: For complex cases, contact the [CMIP International Project Office](mailto:cmip-ipo@esa.int)

Author: Is this the best route. Are we not better off getting people to raise things in public? Or is this for sensitive issues (in which case, should we change the phrasing to the below?)

```suggestion
- **Contact IPO**: For sensitive cases that can't be discussed in public via GitHub, contact the [CMIP International Project Office](mailto:cmip-ipo@esa.int)
```

Reviewer 2: I would be tempted to just go with the following;

```suggestion
- **Contact IPO**: For complex or sensitive cases, contact the [CMIP International Project Office](mailto:cmip-ipo@esa.int)
```

- GitHub: https://github.com/WCRP-CMIP/esgf-vocab
- Documentation: https://esgf.github.io/esgf-vocab/
- Accessing the CMIP7 CVs via esgvoc documentation: https://github.com/WCRP-CMIP/CMIP7-CVs#viewing-the-cvs-and-browsing-terms

### CV Repositories

| Repository | Content |
|------------|---------|
| [CMIP7-CVs](https://github.com/WCRP-CMIP/CMIP7_CVs/) | CMIP7-specific CVs |
| [WCRP-Universe](https://github.com/WCRP-CMIP/WCRP-Universe/) | ESGVOC collection for all WCRP-CMIP projects |
---

### EMD Repositories

These repositories maintain the information used by the EMD.
Some of the information is automatically translated out into the CV repositories,
but these EMD repositories are not themselves CV repositories.

> **Your data is only registered in the CVs if it appears in the CVs repositories above. Registration in the EMD repositories alone is not registration in the CVs.**

Reviewer: ```suggestion
**Your data is only registered in the CVs if it appears in the CVs repositories above. Registration in the EMD repositories alone is not registration in the CVs. However the EMD information required by the CVs should automatically propagate into the CVs on successful completion of the relevant EMD stages.**
```
(to avoid users perhaps thinking they need to do a separate step after completing EMD)

Reviewer 4: Agree with the intent, but the wording should be more definitive. Something like:
```suggestion
  **Your data is only registered in the CVs if it appears in the CVs repositories above. Registration in the EMD repositories alone is not registration
  in the CVs. However, EMD source_id and grid_label are automatically picked up into the CVs upon successful completion of the relevant EMD stages.**
```
